### ISSUE TITLE
Add queue mechanism for long running background processes

### ISSUE THREAD

**Issue by fashxp** [pimcore/pimcore#8374]
for tasks like newsletter sending, video transcoding, web2print generation...

**Comment by ThisIsJustARandomGuy** [comment]
I'd love something similar to the way it's done in Laravel: https://laravel.com/docs/8.x/queues

It's quite flexible and allows for multiple workers running concurrently, rate limiting, locking, queueing of closures, middleware, overlap-less job execution, batches and more. The API is very easy to work with and very expressive. Take a look at this example (copied from the docs):

```<?php

namespace App\Jobs;

use Illuminate\Support\Facades\Redis;

class ProcessPodcast implements ShouldQueue
{
    /**
     * The number of times the job may be attempted.
     *
     * @var int
     */
    public $tries = 25;

    /**
     * The maximum number of unhandled exceptions to allow before failing.
     *
     * @var int
     */
    public $maxExceptions = 3;

    /**
     * Execute the job.
     *
     * @return void
     */
    public function handle()
    {
        Redis::throttle('key')->allow(10)->every(60)->then(function () {
            // Lock obtained, process the podcast...
        }, function () {
            // Unable to obtain lock...
            return $this->release(10);
        });
    }
}

**Comment by aarongerig** [comment]
@ThisIsJustARandomGuy With Symfony this could be done via the Messenger component. WDYT? https://symfony.com/doc/current/components/messenger.html

**Comment by ThisIsJustARandomGuy** [comment]
@aarongerig This seems like a great fit although I'm not very experienced with Symfony itself or the Messenger component yet.

**Comment by dpfaffenbauer** [comment]
@fashxp Symfony Messenger would actually be the way to handle that.

**Comment by brusch** [comment]
see also #6371

**Comment by brusch** [comment]
Closed in favor of #8374
